app/src/main/java/com/safetyheads/akademiaandroida/YouTube/diffUtil/VideoDiffUtil.kt

```diff
@@ -0,0 +1,27 @@
+package com.safetyheads.akademiaandroida.YouTube.diffUtil
+
+import androidx.recyclerview.widget.DiffUtil
+import com.safetyheads.akademiaandroida.YouTube.entities.video.YouTubeVideoDataClass
+
+class VideoDiffUtil(
+    private val oldList: List<YouTubeVideoDataClass>,
+    private val newList: List<YouTubeVideoDataClass>
+) : DiffUtil.Callback() {
+    override fun getOldListSize(): Int {
+        return oldList.size
+    }
+
+    override fun getNewListSize(): Int {
+        return newList.size
+    }
+
+    override fun areItemsTheSame(oldItemPosition: Int, newItemPosition: Int): Boolean {
+        return oldList[oldItemPosition] == newList[newItemPosition]
+    }
+
+    override fun areContentsTheSame(oldItemPosition: Int, newItemPosition: Int): Boolean {
+        val oldVideo = oldList[oldItemPosition]
+        val newVideo = newList[newItemPosition]
+        return oldVideo.items[0].snippet.title == newVideo.items[0].snippet.title
```

Comment:
I think here you should compare id's probably on yt can exist videos in the same chanel/playlist with same name